src/jsx/import/content/register.jsx

```diff
@@ -31,168 +31,169 @@ var RegisterForm = React.createClass({
+                        console.log(`Error Status: ${asynchStatus}`);
+                    }
+                    if (asynchError) {
+                        this.setState({ ajax_fail_error: asynchError });
+                        console.log(`Error Thrown: ${asynchError}`);
+                    }
+                },
+                // pass ajax arguments
+                ajaxArguments,
+            );
         } else {
             if (!this.state.value_username) {
-                this.setState({validated_username: false});
+                this.setState({ validated_username: false });
             }
             if (!this.state.value_email) {
-                this.setState({validated_email: false});
+                this.setState({ validated_email: false });
             }
             if (!this.state.value_password) {
-                this.setState({validated_password: false});
+                this.setState({ validated_password: false });
             }
 
-            const action = setSpinner({'spinner': false});
+            const action = setSpinner({ spinner: false });
             this.props.dispatchSpinner(action);
         }
-    },
-    componentWillMount: function() {
-      // update redux store
-        const action = setLayout({'layout': 'register'});
+    }
+    componentWillMount() {
+        // update redux store
+        const action = setLayout({ layout: 'register' });
         this.props.dispatchLayout(action);
-    },
-    validateUsername: function(event) {
+    }
+    validateUsername(event) {
         const username = event.target.value;
-        const check = checkValidString(username) ? true : false;
+        const check = !!checkValidString(username);
         if (!check) {
-            const action = setSpinner({'spinner': false});
+            const action = setSpinner({ spinner: false });
             this.props.dispatchSpinner(action);
         }
 
-        this.setState({validated_username_server: true});
-        this.setState({validated_username: check});
-        this.setState({value_username: username});
-    },
-    validateEmail: function(event) {
+        this.setState({ validated_username_server: true });
+        this.setState({ validated_username: check });
+        this.setState({ value_username: username });
+    }
+    validateEmail(event) {
         const email = event.target.value;
-        const check = checkValidEmail(email) ? true : false;
+        const check = !!checkValidEmail(email);
         if (!check) {
-            const action = setSpinner({'spinner': false});
+            const action = setSpinner({ spinner: false });
             this.props.dispatchSpinner(action);
         }
 
-        this.setState({validated_email_server: true});
-        this.setState({validated_email: check});
-        this.setState({value_email: email});
-    },
-    validatePassword: function(event) {
+        this.setState({ validated_email_server: true });
```

Comment:
Just a quick question, since I haven't studied much into javascript syntax. But, is it generally preferred to have:

```javascript
this.setState({ validated_email_server: true });
```

Instead of without leading, and trailing space:

```javascript
this.setState({validated_email_server: true});
```

Reply:
I think it's more clear to see that it's a object passed into the function. That's a personal preference based on lint I am used to, though.